elements/rh-dialog/rh-dialog.css

```diff
@@ -74,121 +72,133 @@ section {
   font-size: var(--rh-font-size-heading-sm, 1.5rem);
   font-weight: var(--rh-font-weight-body-text-regular, 400);
-  font-family: 'Red Hat Display', RedHatDisplay, Overpass, Helvetica, sans-serif;
 }
 
 [part='close-button'] {
-  color: var(--rh-color-icon-subtle, #707070);
-  background-color: transparent;
-  border: none;
-  margin: 0;
-  padding: 0;
-  text-align: left;
-  position: absolute;
+  color: var(--rh-color-icon-secondary);
   cursor: pointer;
-  line-height: 24px;
-  padding-block: 0.375rem;
-  padding-inline: var(--rh-space-lg, 16px);
-  top: 0;
-  right: calc(var(--rh-space-xl, 24px) / -3);
+  display: block;
+  inset-block-start: var(--rh-length-xs, 4px);
+  inset-inline-end: var(--rh-length-xs, 4px);
+  position: absolute;
+  z-index: 500;
 }
 
-[part='close-button'] > svg {
-  font-size: 16px;
-  width: var(--rh-space-lg, 16px);
-  aspect-ratio: 1/1;
+[part='close-button']::part(button) {
+  padding: var(--rh-space-xl, 24px);
 }
 
-[part='close-button']:is(:hover, :focus-within, :focus-visible) svg:is(svg, :hover) {
-  fill: var(--rh-color-icon-secondary-on-light, #151515);
+@media (min-width: 1200px) {
+  [part='close-button']::part(button) {
+    padding: var(--rh-space-2xl, 32px);
+  }
 }
 
 :host([position='top']) #dialog {
-  align-self: start;
-  margin-block: var(--rh-space-2xl, 32px);
-  margin-inline: var(--rh-space-lg, 16px);
-  width: 100%;
-  max-width: calc(100% - min(var(--rh-space-2xl, 32px) * 2, var(--rh-space-2xl, 32px)));
-  max-height: calc(100% - var(--_height-offset) - var(--_spacer-align-top));
+  inline-size: 100%;
+  margin-block-start: var(--rh-space-2xl, 32px);
+  max-inline-size: var(--_box-width, calc(100% - var(--rh-space-2xl, 32px)));
 }
 
-footer {
-  display: flex;
+[part='footer'] {
   align-items: center;
+  display: flex;
   gap: var(--rh-space-md, 8px);
 }
 
 #rhds-wrapper {
-  display: contents;
-  font-family: 'Red Hat Text', RedHatText, Overpass, Helvetica, sans-serif;
-
   --offset: var(--rh-space-md, 8px);
   --offset-top: var(--offset);
   --offset-right: var(--offset);
-}
 
-:host([type='video']) {
-  --rh-dialog-close-button-color: var(--rh-color-icon-secondary-on-dark, #ffffff);
+  display: contents;
+  font-family: var(--rh-font-family-body-text, RedHatText, 'Red Hat Text', 'Noto Sans Arabic', 'Noto Sans Hebrew', 'Noto Sans JP', 'Noto Sans KR', 'Noto Sans Malayalam', 'Noto Sans SC', 'Noto Sans TC', 'Noto Sans Thai', Helvetica, Arial, sans-serif);
 }
 
 :host([type='video']) [part='close-button'] {
-  top: var(--offset-top);
-  right: var(--offset-right);
-  padding: var(--rh-space-sm, 6px);
-  color: var(--rh-color-icon-secondary-on-dark, #ffffff);
+  inset-inline-end: var(--offset-right);
+  inset-block-start: var(--offset-top);
+}
+
+:host([type='video']) #close-button::part(button) {
```

Comment:
maybe we should use shadow classes for this instead of attr selectors? wdyt?

Reply:
Resolved in [c292dbe](https://github.com/RedHat-UX/red-hat-design-system/pull/2078/commits/c292dbeb9ee26c02f87c0f4830c2b17caa329922).